src/conn/mod.rs

```diff
@@ -1901,6 +1910,47 @@ mod test {
             assert_eq!(i, 3);
         }
 
+        #[test]
+        fn should_pipeline_work() {
+            let mut conn = Conn::new(get_opts()).unwrap();
+            let stmt1 = conn.prep("select 1 + ?").unwrap();
+            let stmt2 = conn.prep("select 10 + ?").unwrap();
+            let stmt3 = conn.prep("select 100 + ?").unwrap();
+            let mut pipe = conn.pipeline();
+            pipe.exec(&stmt1, (1,)).unwrap();
+            pipe.exec(&stmt2, (2,)).unwrap();
+            pipe.exec(&stmt3, (3,)).unwrap();
+            let mut i = 0;
+            let mut results = pipe.finish();
+            while let Some(result) = results.iter().unwrap() {
+                i += 1;
+                for row in result {
+                    match i {
+                        1 => assert_eq!(row.unwrap().unwrap(), vec![Int(2)]),
+                        2 => assert_eq!(row.unwrap().unwrap(), vec![Int(12)]),
+                        3 => assert_eq!(row.unwrap().unwrap(), vec![Int(103)]),
+                        _ => unreachable!(),
+                    }
+                }
+            }
+        }
+
+        #[test]
+        fn should_pipeline_break_during_results_work() {
```

Comment:
What is this testing? It seems like we executed 2 statements in the pipeline, processed results for one, dropped the rest of the results, and now executing a third statement returns the expected result? If that's the case, the test name doesn't quite line up with that imo.

Reply:
The test is simulating something like this:

```rust
let mut pipeline = conn.pipeline();
// queue up a bunch of queries
let mut results = pipe.finish();
while let Some(result) = results.iter().unwrap() {
  if ... {
    break; // or return, etc.
  }
}
```
It seemed a bit silly to have a loop that called `break` on the first iteration, so I just skipped the loop. (This is the test that fails if `PipelineResults` doesn't implement `Drop`.)